zeebe/engine/src/test/java/io/camunda/zeebe/engine/processing/user/DeleteUserTest.java

```diff
@@ -0,0 +1,89 @@
+/*
+ * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH under
+ * one or more contributor license agreements. See the NOTICE file distributed
+ * with this work for additional information regarding copyright ownership.
+ * Licensed under the Camunda License 1.0. You may not use this file
+ * except in compliance with the Camunda License 1.0.
+ */
+package io.camunda.zeebe.engine.processing.user;
+
+import static io.camunda.zeebe.protocol.record.Assertions.assertThat;
+
+import io.camunda.zeebe.engine.util.EngineRule;
+import io.camunda.zeebe.protocol.record.RejectionType;
+import io.camunda.zeebe.test.util.record.RecordingExporterTestWatcher;
+import java.util.UUID;
+import org.junit.ClassRule;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.jupiter.api.DisplayName;
+
+public class DeleteUserTest {
+  @ClassRule public static final EngineRule ENGINE = EngineRule.singlePartition();
+
+  @Rule
+  public final RecordingExporterTestWatcher recordingExporterTestWatcher =
+      new RecordingExporterTestWatcher();
+
+  @DisplayName("should delete a user if the user exists")
+  @Test
+  public void shouldDeleteAUser() {
+    final var username = UUID.randomUUID().toString();
+    final var userRecord =
+        ENGINE
+            .user()
+            .newUser(username)
+            .withName("Foo Bar")
+            .withEmail("foo@bar.com")
+            .withPassword("password")
+            .create();
+
+    final var deletedUser =
+        ENGINE
+            .user()
+            .deleteUser(userRecord.getKey())
+            .withUsername(userRecord.getValue().getUsername())
+            .withName("Bar Foo")
+            .withEmail("foo@bar.blah")
+            .withPassword("Foo Bar")
+            .delete()
+            .getValue();
+
+    assertThat(deletedUser)
+        .isNotNull()
+        .hasFieldOrPropertyWithValue("username", username)
+        .hasFieldOrPropertyWithValue("name", "Bar Foo")
+        .hasFieldOrPropertyWithValue("email", "foo@bar.blah")
+        .hasFieldOrPropertyWithValue("password", "Foo Bar");
+  }
+
+  @DisplayName("should reject the command if user does not exist")
+  @Test
+  public void shouldRejectTheCommandIfUserDoesNotExist() {
+    final var username = UUID.randomUUID().toString();
+    final var userRecord =
+        ENGINE
```

Comment:
🔧 This can be removed, as you are fetching the user by key. You can just try to delete a user providing a reasonable `userKey` (instead of `-1`)